pkg/git/libgit2/checkout_test.go

```diff
@@ -77,3 +81,48 @@ func TestCheckoutTagSemVer_Checkout(t *testing.T) {
 		t.Errorf("expected semver hash %s, got %s", cTag.Hash(), cSemVer.Hash())
 	}
 }
+
+// This test is specifically to detect regression in libgit2's ED25519 key
+// support.
+// Refer: https://github.com/fluxcd/source-controller/issues/399
+func TestCheckout_ED25519(t *testing.T) {
+	g := NewWithT(t)
+
+	timeout := 5 * time.Second
+	url := "ssh://git@github.com/projectcontour/contour"
```

Comment:
Can the test git server not be used here, rather than relying on the internet? I've used it in the image-automation-controller tests.

Reply:
+1 for getting rid of the references to "real" Git repositories, and replacing them with mocks.

Reply:
The issues with mocks is that we can't tell if our code works on large repos with a consistent history. If we remove it from here please add it to e2e tests, using nodejs would be best as it has many GB in .git.

Reply:
What do you mean by "consistent history" that's lacking in a local server?

Reply:
Though yes, integration tests are the place to test "real world" scenarios. But this particular test is for correctness, so using a local server is not just fine, it's preferred, since it helps isolate the property under test.

Reply:
I should've said "large history", as in hundreds of tags and thousands of commits.

Reply:
That should be handled as a separate test case, most likely with a specific (Go) tag attached to it (e.g. `integration`), so that it only runs in CI.

Reply:
I've refactored the test to use a test git server.